homework/list.js

```diff
@@ -1,79 +1,99 @@
+		document.addEventListener('DOMContentLoaded', this.start.bind(this));
+	}
 
-  var _listNoteContent = [];
-  var _wrapper = document.querySelector('#note-list-wrapper');
+	ListManager.prototype = {
 
-  function start() {
-    fetchList(function(data) {
-      updateList(data);
-      drawList();
-      preloadFirstNote();
-    });
-    window.addEventListener('click', function(event) {
-      onNoteOpen(event);
-    });
-  }
+		/** Initialize the listManager. */
+		start() {
+			this.fetchList()
+				.then(this.updateList.bind(this))
+				.then(this.drawList.bind(this))
+				.then(this.preloadFirstNote.bind(this))
+				.catch(function(error) {
+					console.error(error);
+				});
+			window.addEventListener('click', this.onNoteOpen.bind(this));
+		},
 
-  function onNoteOpen(event) {
-    if (event.target.classList.contains('note-title')) {
-      var id = event.target.dataset.noteId;
-      var content = _listNoteContent[id];
-      window.dispatchEvent(new CustomEvent('note-open',
-        { detail: content }));
-    };
-  }
+		/**
+    * Get List form localhost
+    *
+    * @return {Promise} A start point of the work flow.
+    */
+		fetchList(afterFetch) {
+			return  new Promise(function(resolve, reject) {
+				var req = new XMLHttpRequest();
+				req.open('GET', 'http://127.0.0.1:8000/demo-list-notes.json', true);
+				req.responseType = 'json';
+				req.onload = function () {
+					if (req.status === 200 )
+						resolve(req.response);
+				};
+				req.onerror = function () {
+					reject( new Error(req.statusText));
+				};
+				req.send();
+			});
+		},
 
-  function preloadFirstNote() {
-    if (_listNoteContent.length !== 0) {
-      var content = _listNoteContent[0];
-      window.dispatchEvent(new CustomEvent('note-open',
-        { detail: content }));
-    }
-  }
+    /** Update the list */
+		updateList(list) {
+			this._listNoteContent = list;
+		},
 
-  function updateList(list) {
-    _listNoteContent = list;
-  }
+		/** Append all list title */
+		drawList() {
+			var list = this._listNoteContent;
+			var ul = document.createElement('ul');
+			ul.id = 'note-title-list';
+			var buff = document.createDocumentFragment();
+			list.forEach(insertItem.bind(buff));
+			ul.appendChild(buff);
+			this._wrapper.appendChild(ul);
+		},
 
-  function drawList() {
-    var list = _listNoteContent;
-    var ul = document.createElement('ul');
-    ul.id = 'note-title-list';
-    var buff = document.createDocumentFragment();
-    list.forEach(function(note, i) {
-      var li = document.createElement('li');
-      li.dataset.noteId = i;
-      li.classList.add('note-title');
-      li.textContent = note.title;
-      // Note: buff is captured, so we now have a
-      // little closure naturally.
-      buff.appendChild(li);
-    });
-    ul.appendChild(buff);
-    _wrapper.appendChild(ul);
-  }
+		/** Show first note before user do anything. */
+		preloadFirstNote() {
+			if (this._listNoteContent.length !== 0) {
```

Comment:
To improve expressiveness you should move the condition outside the callee, and put it in the caller. Since there is only one branch of such `if...else`